test/framework/src/main/java/org/elasticsearch/action/support/ActionTestUtils.java

```diff
@@ -65,6 +66,21 @@ public static <Request extends ActionRequest, Response extends ActionResponse> v
         action.execute(task, request, listener);
     }
 
+    public static <Request extends ActionRequest, Response extends ActionResponse> void execute(
+        TransportAction<Request, Response> action,
+        Task task,
+        RequestBuilder<Request> requestBuilder,
+        ActionListener<Response> listener
+    ) {
+        Request request = requestBuilder.buildRequest();
+        try {
+            ActionTestUtils.execute(action, task, request, ActionListener.runAfter(listener, request::decRef));
+        } catch (Exception e) {
+            request.decRef();
+            throw e;
+        }
```

Comment:
Suggest `ActionListener.run(ActionListener.runAfter(listener, request::decRef), l -> ActionTestUtils.execute(action, task, request, l));` since really we should be feeding all exceptions to the listener.

Reply:
Good idea. Thanks.